### PR TITLE
- organize the dbus/roloc constants better

### PR DESCRIPTION
**Pull request by dmacattack**
- remove unecessary code/comments
- add dbus methods to simplify functionality
- fix member variable abuse
- make code more readable and understandable
- fix the startup case 
- use enums more to have type protection

### CODE REVIEW

File: com.inspectron.inspROLOCcontroller.xml [hunk 1]
```diff
@@ -29,6 +29,11 @@
         <arg name="mode" type="i" direction="in"/>
        <arg name="freq" type="i" direction="in"/>
     </method>
-
+    <method name="rolocSetMode">
```
Comment (dmacattack) [inline com.inspectron.inspROLOCcontroller.xml:32]: i added dbus methods, cuz its dumb to have a methods with "no change"
eg: setParameters(no_change, 32.5MHz); 
eg: setParameters(mode_depth, no_change);

File: inspRolocControllerDbus.cpp [hunk 2]
```diff
@@ -84,20 +92,130 @@ void InspROLOCControllerDbus::sendPresent(bool isPresent)
...
+    }
+    else
+    {
+        qWarning() << "illegal volume requested" << lvl << "!!! turn it off !!!";
+        // invalid data. turn the volume off
+        emit setVolumeHandler(ROLOC::eVOLUME_OFF);
+    }
 }
 
+// TODO you shouldnt have to force the same mode being sent. it should be split into two APIs
```
Comment (dmacattack) [inline inspRolocControllerDbus.cpp:107]: I guess i did this. itll get cleaned up on the next commit

File: inspRolocControllerDbus.cpp [hunk 3]
```diff
@@ -84,20 +92,130 @@ void InspROLOCControllerDbus::sendPresent(bool isPresent)
...
+        emit setParametersHandler(rolocMode, rolocFreq);
+    }
+}
+
+/**
+ * @brief InspROLOCControllerDbus::rolocSetMode - dbus slot to set the mode
+ */
+void InspROLOCControllerDbus::rolocSetMode(int mode)
+{
+    qCritical() << "request to set the mode to " << mode; // TODO test and remove
```
Comment (dmacattack) [inline inspRolocControllerDbus.cpp:130]: will be removed next review.

File: inspRolocControllerDbus.cpp [hunk 4]
```diff
@@ -84,20 +92,130 @@ void InspROLOCControllerDbus::sendPresent(bool isPresent)
...
+           (mode == ROLOC::eMODE_BALANCE               ? ROLOC_DBUS_API::eROLOC_MODE_BALANCE               :
+                                                         ROLOC_DBUS_API::eROLOC_MODE_OFF )))));
+}
+
+/**
+ * @brief InspROLOCControllerDbus::getMode - convert dbus API to roloc data
+ */
+ROLOC::eLINEFINDER_MODE InspROLOCControllerDbus::getMode(int mode)
+{
+    return (mode == ROLOC_DBUS_API::eROLOC_MODE_GET_SIGNAL_STRENGTH   ? ROLOC::eMODE_GET_SIGNAL_STRENGTH   :
```
Comment (dmacattack) [inline inspRolocControllerDbus.cpp:203]: all the dbus->roloc  & roloc -> dbus constant conversions are in this class now. dbus constants are out of roloccontroller.cpp

File: inspRolocControllerDbus.hpp [hunk 5]
```diff
@@ -16,37 +17,30 @@ namespace ROLOC_DBUS_API
 
     enum eROLOC_MODE
     {
-        eROLOC_MODE_NOCHANGE,
```
Comment (dmacattack) [inline inspRolocControllerDbus.hpp:19]: no change is dumb

File: inspRolocControllerDbus.cpp [hunk 6]
```diff
@@ -84,20 +92,130 @@ void InspROLOCControllerDbus::sendPresent(bool isPresent)
...
+           (mode == ROLOC_DBUS_API::eROLOC_MODE_GET_DEPTH_MEASUREMENT ? ROLOC::eMODE_GET_DEPTH_MEASUREMENT :
```
Comment (rabidmonkeyman) [inline inspRolocControllerDbus.cpp:204]: how come there are two different constants to express the same thing? i understand one is for dbus and one probly represents the values from the roloc but why not just emit the roloc stuff out of dbus?

File: rolocarrows.cpp [hunk 7]
```diff
@@ -1,21 +1,10 @@
...
-
 /**
  * @brief ROLOCArrows::ROLOCArrows - ctor
  */
 ROLOCArrows::ROLOCArrows()
-: mStatusByte(0x00)
-, mbLeftArrow  (false)
-, mbRightArrow (false)
-, mbCenterArrow(false)
+: mArrow()
```
Comment (dmacattack) [inline rolocarrows.cpp:7]: I simplified this class a lot

**Review by rabidmonkeyman — APPROVED** [review]

File: inspRolocControllerDbus.cpp (hunk 6 above)
Comment (dmacattack) [reply inspRolocControllerDbus.cpp:204]: the DBUS enum is in numerical order, but the ROLOC constants are wildly not.  for example  the frequency constants are wildly not. (0, 0x0800, 0x1000, ....)
looking at the documentation I suspect that these are wrong, so i didnt bother just using one constant.

**Pull request merged**